extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/MultiStageQueryContext.java

```diff
@@ -143,6 +143,13 @@ public class MultiStageQueryContext
 
   public static final String CTX_IS_REINDEX = "isReindex";
 
+  /**
+   * Key for controller task's context passed to worker tasks.
+   * Facilitates sharing the controller's execution environment
+   * and configurations with its associated worker tasks.
+   */
+  public static final String CTX_OF_CONTROLLER = "controllerCtx";
```

Comment:
This should not be configurable. 
We should revert this. Having another configuration parameter does not make sense to me.

Reply:
My thinking on this:
https://github.com/apache/druid/pull/16041#discussion_r1556171737
My follow up work will need to read the Controller's ctx when managing/scheduling Worker task as only Controller task has the full task info (e.g. hard to know if it's an async download task from the Worker task itself), these labels/tags need to be generated based on Controller task then pass on to Worker to guide the task execution.

Reply:
Just pass only the labels/tag to the worker task always. No need to add a new configuration parameter.

Reply:
Yup, after thinking this again, this new param in Worker cxt is better to be removed for now.
Will put up a PR to do it on Monday. Thanks @cryptoe

Reply:
PR submitted https://github.com/apache/druid/pull/16285 @cryptoe